samza-core/src/main/scala/org/apache/samza/diagnostics/DiagnosticsStreamMessage.java

```diff
@@ -155,6 +158,14 @@ public void addProcessorStopEvents(List<ProcessorStopEvent> stopEventList) {
     }
   }
 
+  /**
+   * Add the job's config to the message.
+   * @param config the config to add.
+   */
+  public void addConfig(Map<String, String> config) {
```

Comment:
Nit: The method definition could be 
`public void addConfig(Config config) {
addToMetricsMessage(GROUP_NAME_FOR_DIAGNOSTICS_MANAGER, CONFIG_METRIC_NAME, (Map<String, String>) config);
}`

So that the cast-to-map logic is encapsulated in this class; and not visible to callers.